mne/dipole.py

```diff
@@ -342,40 +343,50 @@ def _fit_Q(fwd_data, whitener, proj_op, B, B2, B_orig, rd):
     return Q, gof, B_residual
 
 
-def _fit_dipoles(data, times, rrs, guess_fwd_svd, fwd_data, whitener,
-                 proj_op, n_jobs):
+def _fit_dipoles(min_dist_to_inner_skull, data, times, rrs, guess_fwd_svd,
+                 fwd_data, whitener, proj_op, n_jobs):
     """Fit a single dipole to the given whitened, projected data"""
     from scipy.optimize import fmin_cobyla
     parallel, p_fun, _ = parallel_func(_fit_dipole, n_jobs)
     # parallel over time points
-    res = parallel(p_fun(B, t, rrs, guess_fwd_svd, fwd_data, whitener, proj_op,
-                         fmin_cobyla)
+    res = parallel(p_fun(min_dist_to_inner_skull, B, t, rrs, guess_fwd_svd,
+                         fwd_data, whitener, proj_op, fmin_cobyla)
                    for B, t in zip(data.T, times))
     pos = np.array([r[0] for r in res])
     amp = np.array([r[1] for r in res])
     ori = np.array([r[2] for r in res])
     gof = np.array([r[3] for r in res]) * 100  # convert to percentage
     residual = np.array([r[4] for r in res]).T
+
     return pos, amp, ori, gof, residual
 
 
-def _fit_dipole(B_orig, t, rrs, guess_fwd_svd, fwd_data, whitener, proj_op,
+def _fit_dipole(min_dist_to_inner_skull, B_orig, t, rrs,
+                guess_fwd_svd, fwd_data, whitener, proj_op,
                 fmin_cobyla):
     """Fit a single bit of data"""
-    logger.info('---- Fitting : %7.1f ms' % (1000 * t,))
     B = np.dot(whitener, B_orig)
+    assert min_dist_to_inner_skull >= 0.
 
     # make constraint function to keep the solver within the inner skull
     if isinstance(fwd_data['inner_skull'], dict):  # bem
         surf = fwd_data['inner_skull']
 
         def constraint(rd):
+
+            dist = _compute_nearest(surf['rr'], rd[np.newaxis, :],
+                                    return_dists=True)[1][0]
+
             if _points_outside_surface(rd[np.newaxis, :], surf, 1)[0]:
-                dist = _compute_nearest(surf['rr'], rd[np.newaxis, :],
-                                        return_dists=True)[1][0]
-                return -dist
-            else:
-                return 1.
+                dist *= -1.
```

Comment:
@jmontoyam @Eric89GXL this is where the big change is. We should not set 1. if inside and negative inside. It should go from 0 to positive values smoothly when doing into the brain. Otherwise fmin_cobyla was not converging.